Packs/CortexAttackSurfaceManagement/Scripts/InferWhetherServiceIsDev/InferWhetherServiceIsDev.py

```diff
@@ -0,0 +1,149 @@
+                 {"Key": "Name", "Source": "AWS", "Value": "ssh-ec2-machine-name"}]
+
+    Returns:
+        bool: whether `observed_key_value_pairs` indicates that the described system is
+            used solely for development (has development indicators and lacks any
+            production indicators)
+    """
+    EXACT_DEV_MATCH = ["dv", "noprod", "np", "ppe"]
+    PARTIAL_DEV_MATCH = ["stg", "stag", "qa", "quality", "test", "tst", "exp", "non_prod",
+                         "non-prod", "nonprod", "nprd", "n-prd", "npe", "npd", "pre_prod",
+                         "pre-prod", "preprod", "pprd", "pov", "proof of value", "poc",
+                         "sbx", "sandbox", "internal", "validation", "lab"]
+    EXACT_PROD_MATCH = ["pr"]
+    PARTIAL_PROD_MATCH = ["prod", "prd", "release", "live"]
+
+    has_dev_indicator = False
+    has_prod_indicator = False
+    for kv_pair in observed_key_value_pairs:
+        if not isinstance(kv_pair, Mapping):
+            demisto.info(f"Ignoring item because it is not a mapping: {kv_pair}")
+        else:
+            if "Key" not in kv_pair or "Value" not in kv_pair:
+                demisto.info(f"Ignoring item because it lacks the keys 'Key' and/or 'Value': {sorted(kv_pair.keys())}")
+            else:
+                key = _canonicalize_string(kv_pair.get("Key", ""))
+                value = _canonicalize_string(kv_pair.get("Value", ""))
+
+                if ("env" in key) or (key in ("stage", "function", "lifecycle", "usage", "tier")):
+                    if (("dev" in value and "devops" not in value)
+                            or ("uat" in value and "prod" not in value)
+                            or any([m == value for m in EXACT_DEV_MATCH])
+                            or any([m in value for m in PARTIAL_DEV_MATCH])):
+                        has_dev_indicator = True
+                    elif (any([m == value for m in EXACT_PROD_MATCH])
+                          or any([m in value for m in PARTIAL_PROD_MATCH])):
+                        has_prod_indicator = True
+
+    if has_dev_indicator and not has_prod_indicator:
+        return True
+    else:
+        return False
+
+
+def is_dev_according_to_classifications(classifications: List[str]) -> bool:
+    """
+    Returns whether any of the classification strings indicate that this service is
+    a development environment. The Xpanse ASM classification strings are a defined
+    vocabulary of facts that are derivable from the publicly observed service without
+    joining on any additional data.
+
+    Args:
+        classifications (List[str]): list of Xpanse ASM classification terms
+            (a defined vocabulary)
+            Example value for `classifications`:
+                ["RdpServer", "SelfSignedCertificate"]
+
+    Returns:
+        bool: whether there is an indication within `classifications` that
+            the described system is used for development
+    """
+    DEV_ENV_CLASSIFICATION = "DevelopmentEnvironment"
```

Comment:
please move this constants out from the method

Reply:
[changing constants placement](https://github.com/demisto/content/pull/23927/commits/17b4613b7a0b742bace2f9d73e4bd8fe85b2afb2)